src/prefect/orion/models/block_documents.py

```diff
@@ -425,11 +425,12 @@ async def update_block_document(
     db: OrionDBInterface,
 ) -> bool:
 
+    merge_existing_data = block_document.merge_existing_data
     current_block_document = await session.get(db.BlockDocument, block_document_id)
     if not current_block_document:
         return False
 
-    update_values = block_document.dict(shallow=True, exclude_unset=True)
+    update_values = block_document.dict(shallow=True, exclude_unset=merge_existing_data)
```

Comment:
I think we'll want to exclude the field `merge_existing_data` from the update values as well